mongoengine/document.py

```diff
@@ -524,12 +527,13 @@ def _save_update(self, doc, save_condition, write_concern):
 
         update_doc = self._get_update_doc()
         if update_doc:
-            upsert = save_condition is None
+            if upsert is None:
+                upsert = save_condition is None
             with set_write_concern(collection, write_concern) as wc_collection:
                 last_error = wc_collection.update_one(
                     select_dict, update_doc, upsert=upsert
                 ).raw_result
-            if not upsert and last_error["n"] == 0:
+            if save_condition is not None and last_error["n"] == 0:
```

Comment:
The fact that we silently ignore it if `upsert=False` is not entirely fixing (#564), I'm a bit hesitating to make it raise an exception because I believe people impacted by #564 will prefer to be aware of the problem

What do you think @erdenezul ?

Reply:
Maybe just adding to the doc string explaining the behavior of upsert=False and that it will silently fail?  You do return if an object was created or not at the end of the method.